apps/analysis_framework/serializers.py

```diff
@@ -289,3 +282,183 @@ def update(self, instance, validated_data):
 
     def get_is_admin(self, analysis_framework):
         return analysis_framework.can_modify(self.context['request'].user)
+
+
+# ------------------ Graphql seriazliers -----------------------------------
+def validate_items_limit(items, limit, error_message='Only %d items are allowed. Provided: %d'):
+    if items:
+        count = len(items)
+        print(count, limit)
+        if count > limit:
+            raise serializers.ValidationError(error_message % (limit, count))
+
+
+class AfWidgetLimit():
+    MAX_SECTIONS_ALLOWED = 5
+    MAX_WIDGETS_ALLOWED_PER_SECTION = 10
+    MAX_WIDGETS_ALLOWED_IN_SECONDARY_TAGGING = 50
```

Comment:
Can we define this in `settings` or here is better?

Reply:
This is only for serializer validation. This is only for bad actor scenarios.